modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheGroupMetricsImpl.java

```diff
@@ -57,6 +59,21 @@
     /** Number of partitions need processed for finished indexes create or rebuilding. */
     private final AtomicLongMetric idxBuildCntPartitionsLeft;
 
+    /** Number of rebalanced partitions left. */
```

Comment:
I think we should track all the metrics in `Demander` class and do not place it here.
The metric registry instance can be obtained on the `Demander` startup.
The `register(String name, LongSupplier supplier, @Nullable String desc) ` method with a lambda expression can be used to track a state of the `RebalanceFuture`.